src/prometheus-stack/tasks.yaml

```diff
@@ -29,3 +29,13 @@ tasks:
             name: app.kubernetes.io/name=prometheus-node-exporter
             namespace: monitoring
             condition: Ready
+  - name: gen-crds
+    actions:
+      - description: Generate servicemonitor types
+        cmd: "npx kubernetes-fluent-client crd https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.1/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml src/pepr/operator/crd/generated/prometheus"
+      - description: Generate podmonitor types
+        cmd: "npx kubernetes-fluent-client crd https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.1/example/prometheus-operator-crd/monitoring.coreos.com_podmonitors.yaml src/pepr/operator/crd/generated/prometheus"
+      - description: Pepr Format
+        cmd: "npx pepr format"
+      - description: "Modify types to fix generated CRDs" # This is hacky but makes generation automatic until https://github.com/defenseunicorns/kubernetes-fluent-client/issues/374 is resolved
+        cmd: "npx ts-node src/prometheus-stack/crdFix.ts"
```

Comment:
Added this task to ensure we are keeping our types up to date or at least have a task to do that. Should be handled by existing renovate match: https://github.com/defenseunicorns/uds-common/blob/main/config/renovate.json5#L54-L63

The typescript hack is pretty gross, but works to modify the CRDs and keep this automated. Also okay with dropping that portion or the entire task until the KFC issue is resolved.

Reply:
Decided to comment this out, pending KFC resolution of the issue.